Frameworks/UIKit/UIPasteboard.mm

```diff
@@ -202,6 +202,97 @@ - (BOOL)containsPasteboardTypes:(NSArray*)pasteboardTypes {
 
 /**
  @Status Caveat
+ @Notes only support one item due to platform limit.
+ */
+- (NSArray<NSDictionary<NSString*, id>*>*)items {
+    WADDataPackageView* wasDataPackageView = _getClipboardContent();
+    id ret = [NSMutableDictionary dictionary];
+
+    if ([wasDataPackageView contains:[WADStandardDataFormats text]]) {
+        [ret setValue:[UIPasteboard _getStringFromDataPackageView:wasDataPackageView] forKey:(NSString*)kUTTypeText];
+    }
+
+    if ([wasDataPackageView contains:[WADStandardDataFormats bitmap]]) {
+        [ret setValue:[UIPasteboard _getImageFromDataPackageView:wasDataPackageView] forKey:(NSString*)kUTTypeImage];
+    }
+
+    if ([wasDataPackageView contains:[WADStandardDataFormats uri]]) {
+        [ret setValue:[UIPasteboard _getURLFromDataPackageView:wasDataPackageView] forKey:(NSString*)kUTTypeURL];
+    }
+
+    NSMutableArray<NSDictionary<NSString*, id>*>* items = [[[NSMutableArray alloc] init] autorelease];
+    if ([ret count] != 0) {
+        [items addObject:ret];
+    }
+
+    return items;
+}
+
+/**
+ @Status Caveat
+ @Notes only support one item
+ */
+- (void)setItems:(NSArray<NSDictionary<NSString*, id>*>*)items {
+    if (![_name isEqualToString:UIPasteboardNameGeneral]) {
+        TraceWarning(TAG, L"Only general pasteboard supported");
+        return;
+    }
+
+    if ([items count] > 0) {
+        if ([items count] > 1) {
+            TraceWarning(TAG, L"Only support first item in items, ignoring the rest");
+        }
+
+        WADDataPackage* dataPackage = [[WADDataPackage make] autorelease];
+        dataPackage.requestedOperation = WADDataPackageOperationCopy;
+        __block BOOL dataSet = NO;
+
+        // enumerate each format and copy the supported one to clipboard
+        NSDictionary<NSString*, id>* item = items[0];
+        [item enumerateKeysAndObjectsUsingBlock:^(NSString* pasteboardType, id data, BOOL* stop) {
+            if ([data isKindOfClass:[NSData class]]) {
+                // data is represented with binary form, let _setData who has NSData to handle it
+                dataSet = [UIPasteboard _setData:pasteboardType withData:data on:dataPackage];
+            } else {
```

Comment:
nothing special. For those three supported types, we can copy them them. but for the others,  we can not. I can change it to the other way to make it look better.

---
In reply to: [87522974](https://github.com/Microsoft/WinObjC/pull/1358#discussion_r87522974) [](ancestors = 87522974)